mongodb/configure.sls

```diff
@@ -0,0 +1,79 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+copy_mongodb_key_file:
+  file.managed:
+    - name: {{ mongodb.cluster_key_file }}
+    - contents: "{{ salt.pillar.get('mongodb:cluster_key') }}"
+    - owner: mongodb
+    - group: mongodb
+    - mode: 0600
+    - require:
+      - pkg: install_packages
+
+stop_mongodb_service:
+  service.dead:
+    - running: True
+    - name: mongodb
+    - enable: True
+    - require:
+      - pkg: install_packages
```

Comment:
Maybe add a comment as to why the service needs to be dead? If you just want the service to restart when you update the config then you can use a `watch_in` parameter on the state that sets the config.